src/lib/snack-bar/snack-bar-container.ts

```diff
@@ -22,17 +43,41 @@ import {MdSnackBarContentAlreadyAttached} from './snack-bar-errors';
   templateUrl: 'snack-bar-container.html',
   styleUrls: ['snack-bar-container.css'],
   host: {
-    'role': 'alert'
-  }
+    'role': 'alert',
+    '[@state]': 'state',
+    '(@state.done)': 'markDestroyed($event)'
+  },
+  animations: [
+    trigger('state', [
+      state(SnackBarState.initial, style({transform: 'translateY(100%)'})),
+      state(SnackBarState.visible, style({transform: 'translateY(0%)'})),
+      state(SnackBarState.complete, style({transform: 'translateY(100%)'})),
+      transition(`${SnackBarState.visible} => ${SnackBarState.complete}`,
+                 animate('250ms cubic-bezier(0.4, 0.0, 1, 1)')),
```

Comment:
If these are the universal curves for entry and exist, can we drop them into constants in something like `core/animation`?